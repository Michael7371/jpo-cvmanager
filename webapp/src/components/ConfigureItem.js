import React, { useState } from 'react'

import './css/ConfigureItem.css'

const ConfigureItem = (props) => {
  const [selected, setSelected] = useState(false)

  const toggleSelect = () => {
    let localSelect = !selected
    props.updateRsu(props.index, localSelect)
    setSelected(localSelect)
  }

  return (
    <div
      id={props.indexList.includes(props.index) ? 'selectedconfigitemdiv' : 'configitemdiv'}
      onClick={() => toggleSelect()}
    >
      <p id="configitemtext">
        <strong>{props.ip}</strong>
      </p>
    </div>
  )
}

export default ConfigureItem
