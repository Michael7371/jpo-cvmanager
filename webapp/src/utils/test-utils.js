/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
