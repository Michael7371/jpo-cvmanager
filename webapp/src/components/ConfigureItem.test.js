import React from 'react'
import { render } from '@testing-library/react'
import ConfigureItem from './ConfigureItem'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<ConfigureItem indexList={[]} />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
