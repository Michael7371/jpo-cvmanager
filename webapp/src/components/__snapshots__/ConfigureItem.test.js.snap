// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    id="configitemdiv"
  >
    <p
      id="configitemtext"
    >
      <strong />
    </p>
  </div>
</div>
`;
